numbat/src/typed_ast.rs

```diff
@@ -122,10 +135,23 @@ pub enum Expression {
         Box<Expression>,
         Type,
     ),
+    /// A special binary operator that has a DateTime as one (or both) of the operands
+    BinaryOperatorForDate(
+        Option<Span>,
+        BinaryOperator,
+        /// LHS must be a DateTime
+        Box<Expression>,
+        /// RHS may be a DateTime, a Time unit, or a String
+        Box<Expression>,
+        Type,
+        /// If true, then this is a date subtration operation
+        bool,
+    ),
     FunctionCall(Span, Span, String, Vec<Expression>, Type),
     Boolean(Span, bool),
     Condition(Span, Box<Expression>, Box<Expression>, Box<Expression>),
     String(Span, Vec<StringPart>),
+    DateTime(Span, chrono::DateTime<chrono::Utc>),
```

Comment:
What does the hardcoded `chrono::Utc` mean here?

Reply:
This means that the `DateTime` object is always storing datetimes with the UTC offset.    When converted into a `Value::DateTime` to be displayed, it stored with the same `DateTime<Utc>` but also includes an offset to be used when pretty printing (see value.rs line 9)